container/doctabs.go

```diff
@@ -0,0 +1,298 @@
+package container
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/layout"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// Declare conformity with Widget interface.
+var _ fyne.Widget = (*DocTabs)(nil)
+
+// DocTabs container is used to display various pieces of content identified by tabs.
+// The tabs contain text and/or an icon and allow the user to switch between the content specified in each TabItem.
+// Each item is represented by a button at the edge of the container.
+//
+// Since: 2.0
+type DocTabs struct {
+	baseTabs
+	CreateTab          func() *TabItem
+	OnClosed           func(*TabItem)
+	OnCloseIntercepted func(*TabItem)
```

Comment:
Intercepting is not an after-the-fact event. The window interface calls it `SetCloseIntercept(func())` - can we better match that? i.e. `SetCloseIntercept()` func or `CloseIntercept` field?

Reply:
Done